Purchases: Update manage purchase design

ryelle opened the pull request:
See #12746, implementing a new design for the Purchases section. This PR implements the new design for the main area of the single purchase view.

![screen shot 2017-05-01 at 1 43 13 pm](https://cloud.githubusercontent.com/assets/541093/25587933/378b5bb0-2e74-11e7-9def-73f3f3c38c1e.png)

There have been no functionality changes, only added data to this page to display the plan info, slight code restructuring, and some linting cleanup.

To test:

1. Visit `/me/purchases`
2. Check a wp.com plan purchase
3. Check a Jetpack plan purchase
4. Check a domain mapping purchase

And any other purchases you may have on your account

rickybanister commented:
This looks great. I noticed a small improvement we can make, but will open a separate ticket.

tyxla commented on client/me/purchases/manage-purchase/index.jsx:
```diff
@@ -254,98 +269,145 @@ const ManagePurchase = React.createClass( {
-	},
+	}
 
-	renderPurchaseDetail() {
```
With these changes we might be losing substantial information about some purchase types. For example, if you purchased a theme, in the detail page you might not see that it's actually a theme:

![](https://cldup.com/UjfHwlbl2X.png)

while previously it was indicating that it was a premium theme and it included a link to the theme page:

![](https://cldup.com/ZgiMl_LrJr.png)

We should make sure to check that with all purchase types.

ryelle replied:
Good catch on the theme details — @rickybanister what do you think?

johnHackworth commented on client/me/purchases/manage-purchase/index.jsx:
```diff
@@ -142,13 +153,17 @@ const ManagePurchase = React.createClass( {
 		const { translate } = this.props;
 
-		if ( ! config.isEnabled( 'upgrades/checkout' ) || ! isRenewable( purchase ) || isExpired( purchase ) || isExpiring( purchase ) || ! getSelectedSite( this.props ) ) {
+		if ( ! config.isEnabled( 'upgrades/checkout' ) ) {
```
why did you divide this condition into two different ones?

tyxla replied:
Maybe we can remove that check? It's currently enabled for all environments.

ryelle replied:
line length 😣  it seemed clearer than breaking the conditions into two lines

johnHackworth replied:
I wouldn't do it ... at least in this PR :)

I mean, we probably should do it, but we shouldn't remove a single case in a separate PR, but doing a different branch that removes all the use cases at once

tyxla commented on client/me/purchases/manage-purchase/index.jsx:
```diff
@@ -254,98 +269,145 @@ const ManagePurchase = React.createClass( {
-			cancelPrivacyProtectionNavItem;
+	renderPlanIcon() {
+		const purchase = getPurchase( this.props );
+		if ( isPlan( purchase ) ) {
```
Do we intend to add icons for more purchase types?

johnHackworth commented:
I left a small comment, but this looks good to me :+1:

ryelle commented on client/me/purchases/manage-purchase/index.jsx:
```diff
@@ -142,13 +153,17 @@ const ManagePurchase = React.createClass( {
+			return null;
+		}
+
+		if ( ! isRenewable( purchase ) || isExpired( purchase ) || isExpiring( purchase ) || ! getSelectedSite( this.props ) ) {
```
Sure, but this section already uses these utilities, and I didn't want to add to this PR by removing them - like you said, out of scope for this PR.

rickybanister commented:
Glad you caught that @tyxla. Let's do a couple things here:

**Domain Purchase**

* Add a descriptor below the title. Example for a domain purchase with privacy protection
* Add a clickable card for editing domain details

![image](https://cloud.githubusercontent.com/assets/437258/25679501/95ce0bfc-301b-11e7-883f-ee08b36c4cf7.png)

![image](https://cloud.githubusercontent.com/assets/437258/25682352/e02f3892-3025-11e7-8953-ab08d7619de6.png)

**Theme Purchase**

* Add an icon for theme purchases. We can use `<Gridicon icon="themes" />` and use 8px rounded corners on the parent element, not the gridicon itself (using a circle cropped the theme icon).
* Use the plan purchase's 'view plan details' link pattern for 'theme details'

![image](https://cloud.githubusercontent.com/assets/437258/25683459/39baf74e-302a-11e7-87f0-095156c76920.png)

ryelle commented:
@rickybanister What do you think of using the plans layout for this related info, like this:

<img width="738" alt="screen shot 2017-05-04 at 1 18 50 pm" src="https://cloud.githubusercontent.com/assets/541093/25716208/4b75e73a-30cc-11e7-82a2-e35c998074e1.png">

ryelle commented:
I swapped the PR labels back to needs review/design review because of https://github.com/Automattic/wp-calypso/pull/13533#issuecomment-299048421, but nothing has changed here yet.

rickybanister commented:
Can we pull theme descriptions to populate that paragraph area? I'm ok with using the plans layout if it won't look blank—the colored border is a bit much without any kind of description there.

ryelle commented:
![screen shot 2017-05-04 at 6 13 07 pm](https://cloud.githubusercontent.com/assets/541093/25727097/59a8508a-30f5-11e7-8d03-26f9d8ff16b6.png)

How's this? Some themes might have only 1-sentence short descriptions, but I think most are about this length.

rickybanister commented:
@ryelle I think that's great, can we still sneak in the small text that mentions 'domain + private registration' or 'premium theme' under the name and above the price? I think that would help as well. But your solution to use the plan header seems great.

ryelle commented:
Sure, I can do that on theme purchases, but it would be repetitive info for domain purchases, see https://github.com/Automattic/wp-calypso/pull/13533#issuecomment-299251681 -- the `.blog Domain Registration` would appear twice (in the header, and as the description)

rickybanister commented:
Perhaps we can ask someone from Spectrum to help contribute a longer description or bit of marketing copy for domain purchases? The short descriptor is helpful for scanning, and the longer one can show the value of the domain or other upgrade?

cc @ranh @breezyskies @michelleweber

ranh commented:
Thanks for the ping @rickybanister 

This is a simple description we've used in the past for domains: 

> Replaces your site's free address (example.wordpress.com) with the domain example.com, making it easier to remember and easier to share.

You may want to check with Marketing about more recent efforts along those lines.

Unrelated note, the use of one-word plan names here feels quite jarring.

https://cloud.githubusercontent.com/assets/541093/25587933/378b5bb0-2e74-11e7-9def-73f3f3c38c1e.png

"Professional" is not a product name. 

It stands out both in the title (next to the price) and in the expiration notice, because it feels like an incomplete sentence: an adjective without a noun. 

As recommended by the style guide, I think the full "Jetpack Professional" would be much more appropriate here.

ryelle commented:
Hm, the purchase object itself only has it as "Professional" (vs the WP.com plans which do say "WordPress.com Business") and this goes back to wp.com code. If this is something that should be consistently done, maybe it should be changed there?

For now, I've added a "Jetpack" prefix to all Jetpack plan names, but I'm not sure if/how this should be handled for translation. Really the ideal would be to add Jetpack to the plan names on wp.com, and they're individually translated. I've also added the purchase type to the header.

<img width="742" alt="screen shot 2017-05-08 at 1 56 26 pm" src="https://cloud.githubusercontent.com/assets/541093/25817720/2886480a-33f6-11e7-8566-65e061021bfc.png">

<img width="742" alt="screen shot 2017-05-08 at 1 55 51 pm" src="https://cloud.githubusercontent.com/assets/541093/25817694/167d965e-33f6-11e7-9889-0be9b82d6698.png">

@ranh Do you know if that text is already displayed anywhere in calypso? The only thing I can find like it is the "Free Domain" tooltip text on the plans page, but that's not really the same thing. Also, do you also have text for domain registration?

sirreal commented on client/me/purchases/manage-purchase/index.jsx:
```diff
@@ -41,19 +43,32 @@ import {
+import {
+	isPersonal,
+	isPremium,
+	isBusiness,
```
Nit: It would be nice to alphabetize these 🙂

rickybanister commented:
@ranh yes, I agree about the plan names, and as @ryelle mentions, that's something that needs to happen in the billing system (or something). 

@ryelle do you think you could wrangle someone else from Luna to handle that as you transition to your new role? I think it does make sense to make sure that all happens in the right place.

ranh commented:
> Do you know if that text is already displayed anywhere in calypso? The only thing I can find like it is the "Free Domain" tooltip text on the plans page, but that's not really the same thing. Also, do you also have text for domain registration

I don't know if this copy is used elsewhere in Calypso, probably not. But I also don't know if we have any other copy to use instead. I know that Marketing are working on domains ATM, so maybe they'll have something better to suggest.

> Hm, the purchase object itself only has it as "Professional" (vs the WP.com plans which do say "WordPress.com Business") and this goes back to wp.com code. If this is something that should be consistently done, maybe it should be changed there?

> @ranh yes, I agree about the plan names, and as @ryelle mentions, that's something that needs to happen in the billing system (or something).

That sounds right. I mentioned it here because this is the first I'm seeing this, but if it's a larger issue I completely agree that it should be handled at the root.

richardmuscat commented:
@rickybanister / @ranh / @ryelle / @johnHackworth -- absolutely agree, we should start labelling them as soon as feasible as:
- Jetpack Personal
- Jetpack Premium
- Jetpack Professional

(As background, the word Jetpack is not their currently because when we kicked these off we didn't, at the time, think we wanted the Jetpack brand to be prominently mentioned on WordPress.com. As you know, that decision has now been reversed.)

johnHackworth commented:
I'm not sure about the implications of changing the plans names. There shouldn't be any, but you know ...  I rather prefer if we merge this PR with @ryelle's fix and then I can take a look at it globally in a different branch

rickybanister commented:
@johnHackworth agreed. If we can sort out the copy issue mentioned above then this is ready to merge. I'd also be ok with introducing a new string for this.

johnHackworth commented:
I think we can ship this as soon as the copy issue mentioned by rick up there is fixed ... :ship: !

rickybanister commented:
New copy sounds great, ship it

klimeryk commented:
Just noting here that this broke renewals - fixed in https://github.com/Automattic/wp-calypso/pull/13984 by @dzver. That's because the `ManagePurchase` component was converted to a class, but that caused `handleRenew` to stop working because class methods are not bound by default. See https://facebook.github.io/react/docs/handling-events.html for more info.
